### PR TITLE
Make async worker methods return Promises

### PR DESCRIPTION
**Pull request by kabentley**
simplifies TS code.

Depends on [#194214](https://bentleycs.visualstudio.com/iModelTechnologies/_git/imodel02/pullrequest/194214)

### CODE REVIEW

File: presentation/backend/src/test/NativePlatform.test.ts [hunk 1]
```diff
@@ -55,22 +55,22 @@ describe("default NativePlatform", () => {
...
   });
 
-  it("calls addon's forceLoadSchemas", async () => {
-    addonMock
-      .setup((x) => x.forceLoadSchemas(moq.It.isAny(), moq.It.isAny()))
-      .callback((_db, cb) => { cb({ result: undefined }); })
-      .verifiable();
-    await nativePlatform.forceLoadSchemas(undefined);
-    addonMock.verifyAll();
+  // it("calls addon's forceLoadSchemas", async () => {
```
Comment (pmconne) [inline presentation/backend/src/test/NativePlatform.test.ts:58]: Are these tests no longer passing?

Comment (grigasp) [reply presentation/backend/src/test/NativePlatform.test.ts]: I put this back in place.

**Review by grigasp — APPROVED** [review]

File: presentation/backend/src/test/NativePlatform.test.ts (hunk 1 above)
Comment (kabentley) [reply presentation/backend/src/test/NativePlatform.test.ts]: thanks @grigasp

**Review by pmconne — APPROVED** [review]

File: core/backend/src/IModelDb.ts [hunk 2]
```diff
@@ -1230,15 +1231,13 @@ export abstract class IModelDb extends IModel {
   }
 
   /** Get the IModel coordinate corresponding to each GeoCoordinate point in the input */
-  public async getIModelCoordinatesFromGeoCoordinates(props: string): Promise<IModelCoordinatesResponseProps> {
-    const resultString = this.nativeDb.getIModelCoordinatesFromGeoCoordinates(props);
-    return JSON.parse(resultString) as IModelCoordinatesResponseProps;
+  public async getIModelCoordinatesFromGeoCoordinates(props: IModelCoordinatesRequestProps): Promise<IModelCoordinatesResponseProps> {
```
Comment (pmconne) [inline core/backend/src/IModelDb.ts:1234]: Document breaking public API change here and below in NextVersion.md.

Comment (kabentley) [reply core/backend/src/IModelDb.ts:1234]: added to NextVersion.

**Review by bbastings — APPROVED** [review]

**Pull request merged**